### PR TITLE
add NumberFormat tryParse and test

### PR DESCRIPTION
**Pull request by utamori**
- add tryParse (wrap Parse)
- add tests

related issue https://github.com/dart-lang/i18n/issues/191

origin pr https://github.com/dart-lang/intl/pull/577

### CODE REVIEW

**Review by mosuem — CHANGES_REQUESTED** [review]
Please add a changelog entry

**Comment by utamori** [comment]
Do I just add it to https://github.com/dart-lang/i18n/blob/main/pkgs/intl/CHANGELOG.md#0181-dev? 
I see that 0.18.1 has already been released.

**Comment by mosuem** [comment]
Sorry about that, #635 should fix this - please wait until it is merged, or rebase onto that branch.

Thanks for going through the hassle of refiling the PRs!

File: pkgs/intl/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,8 @@
+## 0.19.0-dev
```
Comment (mosuem) [inline pkgs/intl/CHANGELOG.md:1]: Please also update the pubspec file to match this, see the result of the `Publish` github action workflow.

**Pull request merged**